configuration_service/configuration_service.go

```diff
@@ -78,61 +96,51 @@ func (service ConfigurationService) IsDaemonProcessingRequests(ctx context.Conte
 
 func (service ConfigurationService) authenticate(prefix string, auth *CallerAuthentication) (err error) {
 
-	//Check if the address passed is the expected authentication address
-	if err = service.checkAuthenticationAddress(auth.UserAddress); err != nil {
-		return err
-	}
-
 	//Check if the Signature is not Expired
 	if err = authutils.CompareWithLatestBlockNumber(big.NewInt(int64(auth.CurrentBlock))); err != nil {
 		return err
 	}
 
+	signerFromMessage, err := authutils.GetSignerAddressFromMessage(service.getMessageBytes(prefix, auth.CurrentBlock), auth.GetSignature())
+	if err != nil {
+		log.Error(err)
+		return err
+	}
 	//Check if the Signature is Valid and Signed accordingly
-	if err = authutils.VerifySigner(service.getMessageBytes(prefix, auth.CurrentBlock),
-		auth.GetSignature(), blockchain.HexToAddress(service.address)); err != nil {
+	if err = service.checkAuthenticationAddress(*signerFromMessage); err != nil {
```

Comment:
Do we not need to verifySigner anymore?

Reply:
We do, just that it's not one address but a list of addresses that we need to loop and check.hence a different method